### DISCUSSION TITLE
Advanced Galaxy Training Wishlist

### DISCUSSION THREAD

**Discussion by hexylena** [galaxyproject/training-material#4246] [Ideas]
We have a fantastic intro to Galaxy section, and some really incredible advanced galaxy tutorials, but there are a lot of features that aren't covered now, but really should be! All of these things would be amazing to work towards for Tapas, if anyone is free.

## Uploading

What we have:
- Two fanastic rule builder tutos
- NCBI wrote a nice one, very specific to their infra tho

What's missing:
- we don't cover uploading collections
- no tutorials yet use the remote data sources which are such a cool feature!
- We could use an FTP tutorial as well.

## Galaxy features

What we have:
- Dave C wrote one on extracting workflows
- I wrote one on searching by extracting it from the hub
- There's a decent one on "understanding the history system"

What's missing: 
- Visualisations
- how to use them for different data types (e.g. brain imaging vs pdb vs tables)
- understanding permissions

## Managing data

What we have:
- Marius wrote one on group tags! Awesome!!
- We have an ok one on collections but it could be improved by discussing upload, apply rules, reshaping tools, etc.
- I wrote one on name tags

What's missing:
- Apply rules tool to retag/restructure a collection
- Reshaping collections (flatten list, etc, everything under Collections Tools)
- using shared data libraries
- how to really use group tags since they're immediately lost, so some concrete examples of how to annotate a collection you've built during a workflow that you want to send to deseq (e.g. with apply rules)

## Analysing

What we have:
- Automating workflows (AMAZING)
- We have a good tutorial on workflow parameters, but it could probably be expanded with more realistic use cases to help people understand how to use it, I've heard either that people didn't understand how to map to their use case, or didn't realise that was what they wanted to solve problem X.
- intermine / jupyter / rstudio all done by community

What's missing: 
- Workflow Reports
- workflow invocations
- a good tutorial on using subworkflows and breaking up workflows to make them modular
- all the tips & tricks of managing workflows (copying steps, updating tools in a WF, adding license/user metadata, what to beware of, how to resolve issues like converting a workflow to a collection type)
- importing WFs from TRS/Dockstore/WFH

**Comment by mvdbeek** [comment]
> What's missing:
> 
> * Workflow Reports
> * workflow invocations
> * a good tutorial on using subworkflows and breaking up workflows to make them modular
> * all the tips & tricks of managing workflows (copying steps, updating tools in a WF, adding license/user metadata, what to beware of, how to resolve issues like converting a workflow to a collection type)
> * importing WFs from TRS/Dockstore/WFH

I will bring that up in todays' IWC meeting, I think this would be great to have
